### DISCUSSION TITLE
When using `useBuiltIns: "usage"`, how does Babel determine whether something is being "used"?

### DISCUSSION THREAD

**Discussion by hibachrach** [babel/babel#15009] [Q&A]
It's unclear to me how Babel actually determines when something needing polyfill is being used. As JS is not statically typed, how would Babel know which function/method is being invoked?

Also asked on StackOverflow: https://stackoverflow.com/questions/73941713/how-does-babel-preset-env-determine-usage-when-using-usebuiltins-usage

**Comment by nyngwang** [comment]
Did you find the solution?

**Comment by zloirock** [comment]
Based on assumptions and as redundant as possible.

For example, we see
```js
throw new AggregateError(foo, bar);
```
where `AggregateError` is not defined in this file - and we inject `AggregateError` polyfill and polyfills of common iterators since `AggregateError` accepts iterables.

If we see
```js
console.log(foo.at(bar));
```
and can't determine `foo` type, we inject polyfills of `Array#at` and `String#at` - even if it's an object with your own `.at` method - it's better to load a polyfill if it's not required than have an error on a lack of feature.
